docs/source/install/Developers.rst

```diff
@@ -0,0 +1,144 @@
+Installation for Developers
+===========================
+
+In order to start developing, clone the QMAP repository using
+
+    .. code-block:: console
+
+        $ git clone --recurse-submodules -j8 https://github.com/cda-tum/qmap
+
+Note the :code:`--recurse-submodules` flag. It is required to also clone all the required submodules. If you happen to forget passing the flag on your initial clone, you can initialize all the submodules by executing :code:`git submodule update --init --recursive` in the main project directory.
+
+A C++ compiler supporting *C++17* and a minimum CMake version of *3.14* is required to build the project.
+
+:code:`boost/program_options >= 1.50` is required for building the commandline applications of the mapping tool.
+
+In order to build the exact mapping tool and for the Python bindings to work, the SMT Solver `Z3 >= 4.8.3 <https://github.com/Z3Prover/z3>`_  has to be installed and the dynamic linker has to be able to find the library. This can be accomplished in a multitude of ways:
```

Comment:
```suggestion
In order to build the exact mapping tool and for the Python bindings to work, the SMT Solver `Z3 >= 4.8.15` <https://github.com/Z3Prover/z3>`_  has to be installed and the dynamic linker has to be able to find the library. This can be accomplished in a multitude of ways:
```
The required version has changed over the past. We simply forgot to update the requirement.

Reply:
Updated it accordingly.